pyro/contrib/gp/models/model.py

```diff
@@ -1,87 +1,193 @@
+    :param str name: Name of this model.
     """
     def __init__(self, X, y, kernel, jitter=1e-6, name=None):
         super(GPModel, self).__init__(name)
         self.set_data(X, y)
         self.kernel = kernel
         self.jitter = jitter
 
-    def set_data(self, X, y=None):
+    def model(self):
         """
-        Sets data for Gaussian Process models.
-
-        :param torch.Tensor X: A 1D or 2D tensor of input data for training.
-        :param torch.Tensor y: A tensor of output data for training with
-            ``y.shape[-1]`` equals to number of data points.
+        A "model" stochastic function. If ``self.y`` is ``None``, this method returns
+        mean and variance of the Gaussian Process prior.
         """
-        if X.dim() > 2:
-            raise ValueError("Expected input tensor of 1 or 2 dimensions, "
-                             "actual dim = {}".format(X.dim()))
-        if y is not None and X.shape[0] != y.shape[-1]:
-            raise ValueError("Expected the number of data inputs equal to the number of data "
-                             "outputs, but got {} and {}.".format(X.shape[0], y.shape[-1]))
-        self.X = X
-        self.y = y
+        raise NotImplementedError
 
-    def model(self):
+    def guide(self):
         """
-        A "model" stochastic method.
+        A "guide" stochastic function to be used in variational inference methods. It
+        also gives posterior information to the method :meth:`forward` for prediction.
         """
         raise NotImplementedError
 
-    def guide(self):
+    def forward(self, Xnew, full_cov=False):
         """
-        A "guide" stochastic method.
+        Computes the mean and covariance matrix (or variance) of Gaussian Process
+        posterior on a test input data :math:`X_{new}`.
+
+        :param torch.Tensor Xnew: A 1D or 2D input data for testing. In 2D case, its
+            second dimension should have the same size as of train input data.
+        :param bool full_cov: A flag to decide if we want to predict full covariance
+            matrix or just variance.
+        :returns: loc and covariance matrix (or variance) of :math:`p(f(X_{new}))`
+        :rtype: tuple(torch.Tensor, torch.Tensor)
         """
         raise NotImplementedError
 
+    def set_data(self, X, y=None):
+        """
+        Sets data for Gaussian Process models.
+
+        Some examples to utilize this method are:
+
+        + Batch training on a sparse variational model:
+
+            >>> Xu = torch.tensor([[1., 0, 2]])  # inducing input
+            >>> likelihood = gp.likelihoods.Gaussian()
+            >>> svgp = gp.models.SparseVariationalGP(X, y, kernel, Xu, likelihood)
+            >>> svi = SVI(svgp.model, svgp.guide, optimizer, "ELBO")
+            >>> batched_X, batched_y = X.split(split_size=10), y.split(split_size=10)
+            >>> for Xi, yi in zip(batched_X, batched_y):
+            >>>     svgp.set_data(Xi, yi)
+            >>>     svi.step()
```

Comment:
nit: fix formatting of loop body (`>>>` -> `...`)

Reply:
whoa, didn't notice that, thanks @eb8680 !